### ISSUE TITLE
allow nested values via name="parent.child"

### ISSUE THREAD

**Issue by TheDirigible** [formkit/formkit#373]
### Describe the feature

allow nested values via name="parent.child"

**Comment by justin-schroeder** [comment]
The way to nest items in FormKit is to use parent elements such as a `group` or a `list` — you can then [access these via dot notation for traversal](https://formkit.com/advanced/core#traversal), but we don't pan on implement dot notation as a declarative syntax for creating structure — only traversing it.

Here's an example of a parent/child:

https://formkit.link/db4828a3ea380166ae25a9b1f5eb95d4

If you have further questions, feel free to jump in and ask on the discord: https://discord.gg/Vhu97pAC76

**Comment by TheDirigible** [comment]
Yes I tried the grouping, but it makes the html way too wordy for complex forms that have a lot of various nested items intermingled.  This is a dealbreaker for me, and is also a downgrade from standard Vue.  I can do this in Vue, so why not similar in formkit? 

 `<input type="text" :value="form.aaa.bbb" />`

**Comment by justin-schroeder** [comment]
If your form is less readable with nesting there is likely a structural problem or your form should be broken out into separate components. Keep in mind FormKit doesn't care how deeply you nest your `<FormKit>` components.

The purpose of this nesting is to create *implicit* structure in your form — groups and lists have zero DOM output, and no bearing on the UI. If you're saying you want to v-model a bunch values to some arbitrary object, you can do that just like in Vue. In otherwords, if your form is totally unstructured, you can make your data totally unstructured too, its just not the way most developers want (or should) craft forms.

So specifically — there's nothing a standard Vue does that FormKit does not do. If you want to v-model some deeply nested object, you can do that in FormKit just fine — I wouldn't recommend it — but you can do it (FormKit uses the `value` prop for initial values only):

https://formkit.link/4b768fd2245a377dbb9024d4e86e6306

Of course you make your own decisions about whether or not you want an architected form solution like FormKit or if some feature is a "deal breaker" — no hard feelings here — we're just a friendly open source package. However, next time if you have questions about *how* to use FormKit, come join us in the discord — its a friendly and pretty helpful place 😀 Good Luck! 👍

**Comment by TheDirigible** [comment]
I'm talking about the readability of the source code.  Which is cleaner?
```
<FormKit type="group" name="a1">
    <FormKit type="text" name="bbb"/>
</FormKit>
<FormKit type="group" name="a2">
    <FormKit type="text" name="bbb"/>
</FormKit>
<FormKit type="group" name="a1">
    <FormKit type="group" name="b1">
       <FormKit type="text" name="ccc"/>
    </FormKit>
</FormKit>
```
or

```
<FormKit type="text" name="a1.bbb"/>
<FormKit type="text" name="a2.bbb"/>
<FormKit type="text" name="a1.b1.ccc"/>

```
This data is not unstructured, but it's a highly complex structure, in which it makes sense in one state in the code, but in a different state to the user.

I can use the v-model solution, but I prefer name/value because I don't want active binding, and I don't want to have to include the form prefix everywhere ("some." in your example).  This is not a question about how to use it, it's a feature request.
Thanks.

**Comment by JohnCampionJr** [comment]
Didn't this use to work this way prior to a recent update?  I just noticed all of my FormKit inputs with "subobject.prop" no longer work.

**Comment by justin-schroeder** [comment]
@JohnCampionJr to have those values be structured by name only has never been a feature not, but if there was some unexpected regression that you can reproduce in the playground I would be more than happy to figure that out.

@TheDirigible i think what you’re asking could be accomplished with a plugin, I might play around a bit and see if something materializes.

**Comment by JohnCampionJr** [comment]
I'm not sure how to recreate it, but I know it worked recently.

I'll see if I can work out what changed.

**Comment by justin-schroeder** [comment]
You can change versions in the playground to check for regression

**Comment by GustavoFenilli** [comment]
> I'm talking about the readability of the source code. Which is cleaner?
> 
> ```
> <FormKit type="group" name="a1">
>     <FormKit type="text" name="bbb"/>
> </FormKit>
> <FormKit type="group" name="a2">
>     <FormKit type="text" name="bbb"/>
> </FormKit>
> <FormKit type="group" name="a1">
>     <FormKit type="group" name="b1">
>        <FormKit type="text" name="ccc"/>
>     </FormKit>
> </FormKit>
> ```
> 
> or
> 
> ```
> <FormKit type="text" name="a1.bbb"/>
> <FormKit type="text" name="a2.bbb"/>
> <FormKit type="text" name="a1.b1.ccc"/>
> ```
> 
> This data is not unstructured, but it's a highly complex structure, in which it makes sense in one state in the code, but in a different state to the user.
> 
> I can use the v-model solution, but I prefer name/value because I don't want active binding, and I don't want to have to include the form prefix everywhere ("some." in your example). This is not a question about how to use it, it's a feature request. Thanks.

Your example is not something that common when creating forms, as grouped inputs most likelly should be in the same UI group when creating the view, so your example should be better constructued as such:

```html
<FormKit type="group" name="a1">
  <FormKit type="text" name="bbb" />

  <FormKit type="group" name="b1">
    <FormKit type="text" name="ccc" />
  </FormKit>
</FormKit>

<FormKit type="group" name="a2">
  <FormKit type="text" name="bbb" />
</FormKit>
```

Also when using this syntax how should you know that something is additive to the group and not to replace instead? you could both interpret as a1 has bbb input and a1 has a b1 group with a ccc group, as a1 had bbb and now a1 has a group with b1 and ccc input.

> ```
> <FormKit type="text" name="a1.bbb"/>
> <FormKit type="text" name="a2.bbb"/>
> <FormKit type="text" name="a1.b1.ccc"/>
> ```

**Comment by JohnCampionJr** [comment]
I stand 100% corrected.   It never did work.  I was thinking of old code which was using Vuelidate which does allow this setup.  

I would love to see the possibility though, but understand if you prefer to not do it this way.

Again, my apologies.  :)

**Comment by justin-schroeder** [comment]
@JohnCampionJr and @TheDirigible here's a simple proof of concept plugin that allows this dot-notation behavior. Would probably need a bit more work, but it generally works just fine. Plugins are awesome 💪

https://formkit.link/1fb2c71e38a39278be05577f10131c4e
